sonar-lits-plugin/src/main/java/com/sonarsource/lits/LitsReport.java

```diff
@@ -0,0 +1,246 @@
+import java.util.Map.Entry;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.commons.lang.StringEscapeUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputComponent;
+import org.sonar.api.batch.fs.InputFile;
+
+public class LitsReport {
+
+  private static final Logger LOG = LoggerFactory.getLogger(LitsReport.class);
+
+  private List<ResourceReport> resourceReports;
+  private int issuesCount;
+
+  public LitsReport(Map<String, Set<ReportedIssue>> differentIssues, FileSystem fileSystem) {
+    this.resourceReports = new ArrayList<>();
+    int n = 0;
+    for (Entry<String, Set<ReportedIssue>> stringSetEntry : differentIssues.entrySet()) {
+      resourceReports.add(new ResourceReport(stringSetEntry.getKey(), stringSetEntry.getValue(), fileSystem));
+      n += stringSetEntry.getValue().size();
+    }
+
+    this.issuesCount = n;
+  }
+
+  public int issuesCount() {
+    return issuesCount;
+  }
+
+  public List<ResourceReport> resourceReports() {
+    return resourceReports;
+  }
+
+
+  public static class ReportedIssue {
+
+    String message;
+    InputComponent inputComponent;
+    String moduleKey;
+    String ruleKey;
+    String componentKey;
+    String severity;
+    int line;
+
+    public ReportedIssue(String ruleKey, String componentKey, int line, String message, String severity, String moduleKey) {
+      this.ruleKey = ruleKey;
+      this.componentKey = componentKey;
+      this.moduleKey = moduleKey;
+      this.line = line;
+      this.message = message;
+      this.severity = severity;
+    }
+
+    public ReportedIssue(String ruleKey, String componentKey, int line, String message, String severity, InputComponent inputComponent) {
+      this.ruleKey = ruleKey;
+      this.componentKey = componentKey;
+      this.inputComponent = inputComponent;
+      this.line = line;
+      this.message = message;
+      this.severity = severity;
+    }
+
+    public String message() {
+      return message;
+    }
+
+    public String ruleKey() {
+      return ruleKey;
+    }
+
+    public String componentKey() {
+      return componentKey;
+    }
+
+    public int line() {
+      return line;
+    }
+
+    public String severity() {
+      return severity;
+    }
+
+    public int key() {
+      return this.hashCode();
+    }
+  }
+
+
+  public static class ResourceReport {
+
+    Set<ReportedIssue> issues;
+
+    // moduleKey:relative/path/to/file.foo
+    String componentKey;
+
+    InputComponent inputComponent;
+
+    boolean isFile;
+
+    private Map<Integer, List<ReportedIssue>> issuesPerLine = new HashMap<>();
+
+    public ResourceReport(String componentKey, Set<ReportedIssue> issues, FileSystem fileSystem) {
+      this.componentKey = componentKey;
+      this.issues = issues;
+
+      // InputFile corresponding to this resource, it's null for new issues
+      InputComponent inputComponent = null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "inputComponent" which hides the field declared at line 127. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AHiddenFieldCheck)